internal/release/promote/pr_rendering.go

```diff
@@ -0,0 +1,200 @@
+package promote
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/nestoca/joy/api/v1alpha1"
+	"golang.org/x/mod/semver"
+)
+
+type ChangeType string
+
+const (
+	ChangeTypeUpgrade   ChangeType = "Upgrade"
+	ChangeTypeDowngrade ChangeType = "Downgrade"
+	ChangeTypeUpdate    ChangeType = "Update"
+)
+
+type ReleaseInfo struct {
+	Name         string
+	Project      *v1alpha1.Project
+	CodeOwners   []string
+	Repository   string
+	Source       ReleaseWithGitTag
+	Target       ReleaseWithGitTag
+	OlderGitTag  string
+	NewerGitTag  string
+	IsPrerelease bool
+	// IsUpgrade is now deprecated and kept only temporarily while we transition to ChangeType
+	IsUpgrade  bool
+	ChangeType ChangeType
+	Commits    []*CommitInfo
+	Error      error
+}
+
+type PromotionInfo struct {
+	SourceEnvironment *v1alpha1.Environment
+	TargetEnvironment *v1alpha1.Environment
+	Releases          []*ReleaseInfo
+	Error             error
+}
+
+type CommitInfo struct {
+	Sha          string
+	ShortSha     string
+	Author       string
+	GitHubAuthor string
+	Message      string
+	ShortMessage string
+}
+
+func getReleaseInfo(sourceRelease, targetRelease *v1alpha1.Release, opts PerformOpts) *ReleaseInfo {
+	getAndPrintReleaseInfoWithError := func(msg string, args ...any) *ReleaseInfo {
+		err := fmt.Errorf("Failed to get info for release "+sourceRelease.Name+": "+msg, args...)
+		fmt.Printf("⚠️ %v\n", err)
+		return &ReleaseInfo{
+			Name:  sourceRelease.Name,
+			Error: err,
+		}
+	}
+
+	project := sourceRelease.Project
+	changeType := ChangeTypeUpgrade
+	if targetRelease != nil {
+		comparison := semver.Compare("v"+sourceRelease.Spec.Version, "v"+targetRelease.Spec.Version)
+		if comparison < 0 {
+			changeType = ChangeTypeDowngrade
+		} else if comparison > 0 {
+			changeType = ChangeTypeUpgrade
+		} else {
+			changeType = ChangeTypeUpdate
+		}
+	}
+
+	isPrerelease := false
+	if semver.Build("v"+sourceRelease.Spec.Version) != "" ||
+		semver.Prerelease("v"+sourceRelease.Spec.Version) != "" ||
+		(targetRelease != nil && (semver.Build("v"+targetRelease.Spec.Version) != "" || semver.Prerelease("v"+targetRelease.Spec.Version) != "")) {
+		isPrerelease = true
```

Comment:
Same here for target version
if semver.Build(version) + semver.Prerelease(version) != ""

Reply:
Might as well wrap this into a function `isPrerelease`.